src/client/components/series-tile/series-tiles-row.tsx

```diff
@@ -140,28 +150,52 @@ export class SeriesTilesRow extends React.Component<SeriesTilesRowProps, SeriesT
   drop = (e: React.DragEvent<HTMLElement>) => {
     if (!this.canDrop()) return;
     e.preventDefault();
-    const { clicker, essence: { series } } = this.props;
-
     this.setState({ dragPosition: null });
 
-    const newSeries: Series = DragManager.isDraggingSeries() ? DragManager.draggingSeries() : MeasureSeries.fromMeasure(DragManager.draggingMeasure());
-    if (!newSeries) return;
-
-    let dragPosition = this.calculateDragPosition(e);
+    if (DragManager.isDraggingSeries()) {
+      this.rearrangeSeries(DragManager.draggingSeries(), this.calculateDragPosition(e));
+    } else {
+      this.dropNewSeries(fromMeasure(DragManager.draggingMeasure()), this.calculateDragPosition(e));
+    }
+  }
 
-    if (dragPosition.replace === this.maxItems()) {
-      dragPosition = new DragPosition({ insert: dragPosition.replace });
+  private dropNewSeries(newSeries: Series, dragPosition: DragPosition) {
+    const { clicker, essence: { series } } = this.props;
+    const isSeriesValid = newSeries instanceof MeasureSeries || (newSeries instanceof QuantileSeries && !series.hasSeries(newSeries));
```

Comment:
Change the condition to `if (isDuplicated)`, invert `if` and `else` blocks. It's gonna be more readable, hopefully.